haas/test_common.py

```diff
@@ -242,8 +242,14 @@ def site_layout():
         api.switch_register(**switch)
 
     for node in layout['nodes']:
-        api.node_register(node['name'], node['ipmi']['host'],
-            node['ipmi']['user'], node['ipmi']['pass'])
+        api.node_register(node['name'],
```

Comment:
You shouldn't be passing in the individual fields from the obm object, just the whole thing, e.g. `api.node_register(node['name'], obm=node['obm']). This will prevent it from working with any other OBM driver that doesn't have exactly the same fields. Did I just miss this when reviewing your patch the first time, or did this go in as part of fixing the tests afterwards?